### PR TITLE
Use attribute docstrings for options

### LINKED ISSUE #1762: Use attribute docstrings for options
**What is the expected feature or enhancement?**

Today most of the options dataclasses use the same Google style argument docstring as the rest of the modules. Unfortunately, the rendered API reference would list each field under both `Parameters` and `Attributes`, [for example](https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/qiskit_ibm_runtime.options.EstimatorOptions):

image

image

<br></br>

The one exception is [SamplerExecutionOptions](https://github.com/Qiskit/qiskit-ibm-runtime/blob/3026f9f83b23993607a5aa66e481bbaafdfee521/qiskit_ibm_runtime/options/sampler_execution_options.py#L20), which only has attribute docstring. The generated doc therefore only shows the fields under `Attributes`:

image

<br></br>

I think this is a much nicer format, as you get both the description and type in one place. 

@Eric-Arellano please let us know if you have better ideas! 

**Acceptance criteria**

- Use attribute docstring instead of argument docstring for options classes. 

Also note that when I played with `SamplerExecutionOptions`, the text `Default: True` was not formatted correctly, so I don't know if there is special convention on specifying defaults.

### PR DESCRIPTION
**Pull request by kt474**
### Details and comments
Fixes #1762

### CODE REVIEW

File: qiskit_ibm_runtime/options/dynamical_decoupling_options.py [hunk 1]
```diff
@@ -19,32 +19,30 @@
...
-            as late as ("alap") possible. Default: "alap".
-    """
+    """Options for dynamical decoupling (DD)."""
 
     enable: Union[UnsetType, bool] = Unset
+    r"""Whether to enable DD as specified by the other options in this class.
+        Default: False.
+    """
     sequence_type: Union[UnsetType, Literal["XX", "XpXm", "XY4"]] = Unset
+    r"""Which dynamical decoupling sequence to use. Default: "XX".
```
Comment (SamFerracin) [inline qiskit_ibm_runtime/options/dynamical_decoupling_options.py:29]: default?

I think a bunch of them are missing, so check the other docstrings

Comment (kt474) [reply qiskit_ibm_runtime/options/dynamical_decoupling_options.py:29]: I need to double check these all render correctly too

File: qiskit_ibm_runtime/options/estimator_options.py [hunk 2]
```diff
@@ -34,86 +34,89 @@
...
-            Default: 0.
-
-        resilience_level: How much resilience to build against errors.
-            Higher levels generate more accurate results,
-            at the expense of longer processing times.
+    # Sadly we cannot use pydantic's built in validation because it won't work on Unset.
+    default_precision: Union[UnsetType, float] = Unset
+    r"""The default precision to use for any PUB or ``run()``
+        call that does not specify one.
+        Each estimator pub can specify its own precision. If the ``run()`` method
```
Comment (abbycross) [inline qiskit_ibm_runtime/options/estimator_options.py:43]:
```suggestion
        Each Estimator PUB can specify its own precision. If the ``run()`` method
```

File: qiskit_ibm_runtime/options/estimator_options.py [hunk 3]
```diff
@@ -34,86 +34,89 @@
...
-            <https://docs.quantum.ibm.com/guides/configure-error-mitigation>`_.
-            for more information about the error mitigation methods used at each level.
+        A configuration is a combination of a specific parameter value binding set and a
+        physical measurement basis. A physical measurement basis groups together some
+        collection of qubit-wise commuting observables for some specific circuit/parameter
+        value set to create a single measurement with basis rotations that is inserted into
+        hardware executions.
 
-            Default: 1.
+        If twirling is enabled, the value of this option will be divided over circuit,
```
Comment (abbycross) [inline qiskit_ibm_runtime/options/estimator_options.py:61]: "divided over circuit, randomizations," reads oddly to me but maybe it is codespeak?

File: qiskit_ibm_runtime/options/estimator_options.py [hunk 4]
```diff
@@ -34,86 +34,89 @@
...
-        execution: Execution time options. See :class:`ExecutionOptionsV2` for all available options.
+        * 0: No mitigation.
+        * 1: Minimal mitigation costs. Mitigate error associated with readout errors.
+        * 2: Medium mitigation costs. Typically reduces bias in estimators but
+            is not guaranteed to be zero bias.
 
-        twirling: Pauli twirling options. See :class:`TwirlingOptions` for all available options.
+        Refer to the
+        `Configure error mitigation for Qiskit Runtime
+        <https://docs.quantum.ibm.com/guides/configure-error-mitigation>`_.
```
Comment (abbycross) [inline qiskit_ibm_runtime/options/estimator_options.py:89]:
```suggestion
        <https://docs.quantum.ibm.com/guides/configure-error-mitigation>`_ guide
```

File: qiskit_ibm_runtime/options/estimator_options.py [hunk 5]
```diff
@@ -34,86 +34,89 @@
...
+        Default: ``None``.
+    """
     dynamical_decoupling: Union[DynamicalDecouplingOptions, Dict] = Field(
         default_factory=DynamicalDecouplingOptions
     )
+    r"""Suboptions for dynamical decoupling. See
+        :class:`DynamicalDecouplingOptions` for all available options.
+    """
     resilience: Union[ResilienceOptionsV2, Dict] = Field(default_factory=ResilienceOptionsV2)
+    r"""Advanced resilience options to fine tune the resilience strategy.
```
Comment (abbycross) [inline qiskit_ibm_runtime/options/estimator_options.py:106]:
```suggestion
    r"""Advanced resilience options to fine-tune the resilience strategy.
```

File: qiskit_ibm_runtime/options/layer_noise_learning_options.py [hunk 6]
```diff
@@ -33,35 +33,42 @@ class LayerNoiseLearningOptions:
...
-            would correspond to 6 layers of the layer of interest, for example.
-            Default: (0, 1, 2, 4, 16, 32).
     """
 
     max_layers_to_learn: Union[UnsetType, int, None] = Unset
+    r"""The max number of unique layers to learn.
+        A ``None`` value indicates that there is no limit.
+        If there are more unique layers present, then some layers will not be learned or
+        mitigated. The learned layers are prioritized based on the number of times they
+        occur in a set of run estimator PUBs, and for equally occurring layers are
```
Comment (abbycross) [inline qiskit_ibm_runtime/options/layer_noise_learning_options.py:43]:
```suggestion
        occur in a set of run Estimator PUBs, and for equally occurring layers are
```

File: qiskit_ibm_runtime/options/resilience_options.py [hunk 7]
```diff
@@ -36,56 +36,57 @@
...
-            Default: False.
-
-        pec: Additional probabalistic error cancellation mitigation options.
-            See :class:`PecOptions` for all options.
-
-        layer_noise_learning: Layer noise learning options.
-            See :class:`LayerNoiseLearningOptions` for all options.
+    measure_mitigation: Union[UnsetType, bool] = Unset
+    r"""Whether to enable measurement error mitigation method.
+        If you enable measurement mitigation, you can fine tune its noise learning
```
Comment (abbycross) [inline qiskit_ibm_runtime/options/resilience_options.py:43]:
```suggestion
        If you enable measurement mitigation, you can fine-tune its noise learning
```

File: qiskit_ibm_runtime/options/resilience_options.py [hunk 8]
```diff
@@ -36,56 +36,57 @@
...
-    measure_mitigation: Union[UnsetType, bool] = Unset
     measure_noise_learning: Union[MeasureNoiseLearningOptions, Dict] = Field(
         default_factory=MeasureNoiseLearningOptions
     )
+    r"""Additional measurement noise learning options.
+        See :class:`MeasureNoiseLearningOptions` for all options.
+    """
     zne_mitigation: Union[UnsetType, bool] = Unset
+    r"""Whether to turn on Zero Noise Extrapolation error mitigation method.
+        If you enable ZNE, you can fine tune its options by using :attr:`~zne`.
```
Comment (abbycross) [inline qiskit_ibm_runtime/options/resilience_options.py:57]:
```suggestion
        If you enable ZNE, you can fine-tune its options by using :attr:`~zne`.
```

File: qiskit_ibm_runtime/options/resilience_options.py [hunk 9]
```diff
@@ -36,56 +36,57 @@
...
+
+        Default: False.
+    """
     zne: Union[ZneOptions, Dict] = Field(default_factory=ZneOptions)
+    r"""Additional zero noise extrapolation mitigation options.
+        See :class:`ZneOptions` for all options.
+    """
     pec_mitigation: Union[UnsetType, bool] = Unset
+    r"""Whether to turn on Probabilistic Error Cancellation error mitigation method.
+        If you enable PEC, you can fine tune its options by using :attr:`~pec`.
```
Comment (abbycross) [inline qiskit_ibm_runtime/options/resilience_options.py:68]:
```suggestion
        If you enable PEC, you can fine-tune its options by using :attr:`~pec`.
```

File: qiskit_ibm_runtime/options/resilience_options.py [hunk 10]
```diff
@@ -36,56 +36,57 @@
...
+        See :class:`PecOptions` for additional PEC related options.
```
Comment (abbycross) [inline qiskit_ibm_runtime/options/resilience_options.py:69]:
```suggestion
        See :class:`PecOptions` for additional PEC-related options.
```

File: qiskit_ibm_runtime/options/resilience_options.py [hunk 11]
```diff
@@ -36,56 +36,57 @@
...
-
-    measure_mitigation: Union[UnsetType, bool] = Unset
     measure_noise_learning: Union[MeasureNoiseLearningOptions, Dict] = Field(
         default_factory=MeasureNoiseLearningOptions
     )
+    r"""Additional measurement noise learning options.
+        See :class:`MeasureNoiseLearningOptions` for all options.
+    """
     zne_mitigation: Union[UnsetType, bool] = Unset
+    r"""Whether to turn on Zero Noise Extrapolation error mitigation method.
```
Comment (abbycross) [inline qiskit_ibm_runtime/options/resilience_options.py:56]:
```suggestion
    r"""Whether to turn on Zero-Noise Extrapolation error mitigation method.
```

File: qiskit_ibm_runtime/options/resilience_options.py [hunk 12]
```diff
@@ -36,56 +36,57 @@
...
+        If you enable ZNE, you can fine tune its options by using :attr:`~zne`.
+        See :class:`ZneOptions` for additional ZNE related options.
+
+        Default: False.
+    """
     zne: Union[ZneOptions, Dict] = Field(default_factory=ZneOptions)
+    r"""Additional zero noise extrapolation mitigation options.
```
Comment (abbycross) [inline qiskit_ibm_runtime/options/resilience_options.py:63]:
```suggestion
    r"""Additional zero-noise extrapolation mitigation options.
```

File: qiskit_ibm_runtime/options/resilience_options.py [hunk 13]
```diff
@@ -36,56 +36,57 @@
...
     layer_noise_learning: Union[LayerNoiseLearningOptions, Dict] = Field(
         default_factory=LayerNoiseLearningOptions
     )
+    r"""Layer noise learning options.
+        See :class:`LayerNoiseLearningOptions` for all options.
+    """
     layer_noise_model: Union[UnsetType, NoiseLearnerResult, Sequence[LayerError]] = Unset
+    r"""A :class:`NoiseLearnerResult` or a sequence of :class:`LayerError`
+        objects. If set, all the mitigation strategies that require noise data (e.g., PEC and
+        PEA) skip the noise learning stage, and instead gather the required information from
```
Comment (abbycross) [inline qiskit_ibm_runtime/options/resilience_options.py:86]:
```suggestion
        PEA) skip the noise-learning stage, and instead gather the required information from
```

File: qiskit_ibm_runtime/options/simulator_options.py [hunk 14]
```diff
@@ -37,26 +37,32 @@ class SimulatorOptions:
...
+    """
     seed_simulator: Union[UnsetType, int] = Unset
+    r"""Random seed to control sampling. 
+    
+        Default: ``None``.
+    """
     coupling_map: Union[UnsetType, List[List[int]], CouplingMap] = Unset
+    r"""Directed coupling map to target in mapping. If
+        the coupling map is symmetric, both directions need to be specified.
+        Each entry in the list specifies a directed two-qubit interactions,
```
Comment (abbycross) [inline qiskit_ibm_runtime/options/simulator_options.py:55]:
```suggestion
        Each entry in the list specifies a directed two-qubit interaction,
```

**Review by abbycross — APPROVED** [review]

**Review by Eric-Arellano — APPROVED** [review]
Good change!! I didn't review the specific wording, but the Sphinx setup looks good.

**Pull request merged**